examples/vault-ec2-auth/user-data-vault.sh

```diff
@@ -0,0 +1,87 @@
+for i in $(seq 1 20); do
+  server_output=$(/opt/vault/bin/vault operator init) && exit_status=0 || exit_status=$?
+  if [[ $exit_status -eq 0 ]]; then break; fi
+  sleep 5
+done; (exit $exit_status)
+
+# The expected output should be similar to this:
+# ==========================================================================
+# Unseal Key 1: ddPRelXzh9BdgqIDqQO9K0ldtHIBmY9AqsTohM6zCRl7
+# Unseal Key 2: liSgypzdVrAxz73KbKyCMjVeSnRMuxCZMk1PWIZdjENS
+# Unseal Key 3: pmgeVu/fs8+jl8bOzf3Cq56BFufm4o7Sxt2oaUcvt6Dp
+# Unseal Key 4: i3W2xJEyUqUqcO1QSjTA+Ua0RUPxnNWM27AqaC8wW7Zh
+# Unseal Key 5: vHsQtCRgfblPeFYw1hhCVbji0MoNUP8zyIWhLWs3PebS
+#
+# Initial Root Token: cb076fc1-cc1f-6766-795f-b3822ba1ac57
+#
+# Vault initialized with 5 key shares and a key threshold of 3. Please securely
+# distribute the key shares printed above. When the Vault is re-sealed,
+# restarted, or stopped, you must supply at least 3 of these keys to unseal it
+# before it can start servicing requests.
+#
+# Vault does not store the generated master key. Without at least 3 key to
+# reconstruct the master key, Vault will remain permanently sealed!
+#
+# It is possible to generate new unseal keys, provided you have a quorum of
+# existing unseal keys shares. See "vault operator rekey" for more information.
+# ==========================================================================
+
+# Unseals the server with 3 keys from this output
+# Please note that this is not how it should be done in production as it is not secure
+# Ideally it should be auto unsealed https://www.vaultproject.io/docs/enterprise/auto-unseal/index.html
+# For this quick example specifically, we are just running one vault server and unsealing it like this
+# for simplicity as this example focuses on auth and not on auto unsealing
+echo "$server_output" | head -n 3 | awk '{ print $4; }' | xargs -l /opt/vault/bin/vault operator unseal
+
+# Exports the client token environment variable necessary for running the following vault commands
+export VAULT_TOKEN=$(echo "$server_output" | head -n 7 | tail -n 1 | awk '{ print $4; }')
+
+# Enables AWS authentication
+/opt/vault/bin/vault auth enable aws
+
+# Creates a policy that allows writing and reading from an "example_" prefix at "secret" backend
+/opt/vault/bin/vault policy write "example-policy" -<<EOF
+path "secret/example_*" {
+  capabilities = ["create", "read"]
+}
+EOF
+
+# Creates an authentication role
+# The role name & ami id are being passed by terraform
+# This example uses the ami id as a criteria for whitelisting, but there are multiple
+# other settings you can pick for EC2 metadata auth.
+# Read more at: https://www.vaultproject.io/api/auth/aws/index.html#create-role
+/opt/vault/bin/vault write \
+  auth/aws/role/${example_role_name}\
+  auth_type=ec2 \
+  policies=example-policy \
+  max_ttl=500h \
+  bound_ami_id=${ami_id}
+
+# Writes some secret, this secret is being written by terraform for test purposes
```

Comment:
Again, add a warning that you'd never pass the secret like this in plain text, and that we are only doing it so we can easily run automated tests in this example.